### ISSUE TITLE
MKL missing when installing numpy

### ISSUE THREAD

**Issue by rcthomas** [ContinuumIO/anaconda-issues#9519]
### Actual Behavior

Grab the latest installer, run it, activate base env, try to install numpy --- apparently MKL is not installed anymore?

```
> wget https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
> bash Miniconda3-latest-MacOSX-x86_64.sh -b -p env
> source env/bin/activate
> conda install numpy
Solving environment: done

## Package Plan ##

  environment location: /Users/rthomas/tmp99/env

  added / updated specs:
    - numpy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    blas-1.0                   |         openblas          48 KB
    numpy-1.14.3               |   py36he6379a5_2          42 KB
    libgfortran-3.0.1          |       h93005f0_2         495 KB
    numpy-base-1.14.3          |   py36h9797aa9_2         3.9 MB
    conda-4.5.4                |           py36_0         1.0 MB
    libopenblas-0.2.20         |       hdc02c5d_7         9.9 MB
    ------------------------------------------------------------
                                           Total:        15.4 MB

The following NEW packages will be INSTALLED:

    blas:        1.0-openblas
    libgfortran: 3.0.1-h93005f0_2
    libopenblas: 0.2.20-hdc02c5d_7
    numpy:       1.14.3-py36he6379a5_2
    numpy-base:  1.14.3-py36h9797aa9_2

The following packages will be UPDATED:

    conda:       4.5.1-py36_0          --> 4.5.4-py36_0

Proceed ([y]/n)?
```

### Expected Behavior

Installation of numpy with MKL.
 
### Steps to Reproduce

See above.

##### Anaconda or Miniconda version:

See above.

##### Operating System:

Tested Linux and Mac OS with latest installer

##### `conda info`
```
     active environment : base
    active env location : /Users/rthomas/tmp99/env
            shell level : 1
       user config file : /Users/rthomas/.condarc
 populated config files :
          conda version : 4.5.1
    conda-build version : not installed
         python version : 3.6.5.final.0
       base environment : /Users/rthomas/tmp99/env  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/osx-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /Users/rthomas/tmp99/env/pkgs
                          /Users/rthomas/.conda/pkgs
       envs directories : /Users/rthomas/tmp99/env/envs
                          /Users/rthomas/.conda/envs
               platform : osx-64
             user-agent : conda/4.5.1 requests/2.18.4 CPython/3.6.5 Darwin/15.6.0 OSX/10.11.6
                UID:GID : 32331:1032331
             netrc file : /Users/rthomas/.netrc
           offline mode : False
```

##### `conda list --show-channel-urls`
```
# packages in environment at /Users/rthomas/tmp99/env:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                   py36_0    defaults
blas                      1.0                    openblas    defaults
ca-certificates           2018.03.07                    0    defaults
certifi                   2018.4.16                py36_0    defaults
cffi                      1.11.5           py36h342bebf_0    defaults
chardet                   3.0.4            py36h96c241c_1    defaults
conda                     4.5.4                    py36_0    defaults
conda-env                 2.6.0                h36134e3_0    defaults
cryptography              2.2.2            py36h1de35cc_0    defaults
idna                      2.6              py36h8628d0a_1    defaults
libcxx                    4.0.1                h579ed51_0    defaults
libcxxabi                 4.0.1                hebd6815_0    defaults
libedit                   3.1                  hb4e282d_0    defaults
libffi                    3.2.1                h475c297_4    defaults
libgfortran               3.0.1                h93005f0_2    defaults
libopenblas               0.2.20               hdc02c5d_7    defaults
ncurses                   6.0                  hd04f020_2    defaults
numpy                     1.14.3           py36he6379a5_2    defaults
numpy-base                1.14.3           py36h9797aa9_2    defaults
openssl                   1.0.2o               h26aff7b_0    defaults
pip                       9.0.3                    py36_0    defaults
pycosat                   0.6.3            py36hee92d8f_0    defaults
pycparser                 2.18             py36h724b2fc_1    defaults
pyopenssl                 17.5.0           py36h51e4350_0    defaults
pysocks                   1.6.8                    py36_0    defaults
python                    3.6.5                hc167b69_0    defaults
python.app                2                        py36_8    defaults
readline                  7.0                  hc1231fa_4    defaults
requests                  2.18.4           py36h4516966_1    defaults
ruamel_yaml               0.15.35          py36h1de35cc_1    defaults
setuptools                39.0.1                   py36_0    defaults
six                       1.11.0           py36h0e22d5e_1    defaults
sqlite                    3.23.1               hf1716c9_0    defaults
tk                        8.6.7                h35a86e2_3    defaults
urllib3                   1.22             py36h68b9469_0    defaults
wheel                     0.31.0                   py36_0    defaults
xz                        5.2.3                h727817e_4    defaults
yaml                      0.1.7                hc338f04_2    defaults
zlib                      1.2.11               hf3cbc9b_2    defaults
```

**Comment by msarahan** [comment]
please install blas=*=mkl to remedy this.  Sorry for the confusion.  We're working on better ways to select package variants, and we're having some growing pains.

**Comment by msarahan** [comment]
This is fixed in the latest numpy 1.14.4 and 1.14.5 builds.  Please try again.
